versioned/mongodb/src/main/java/com/dremio/nessie/versioned/impl/condition/MongoDBConditionAliasVisitor.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright (C) 2020 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.dremio.nessie.versioned.impl.condition;
+
+import java.util.List;
+
+import com.dremio.nessie.versioned.store.Entity;
+import com.google.common.collect.ImmutableList;
+
+/*
+ * This uses the Visitor design pattern to retrieve object attributes.
+ */
+public class MongoDBConditionAliasVisitor implements ConditionAliasVisitor {
+  /**
+   * The callback method in the visitor design pattern.
+   * @param conditionExpression The object to be aliased.
+   * @param collector the class that does the aliasing.
+   * @return the aliased ConditionExpression.
+   */
+  @Override
+  public ConditionExpression visit(ConditionExpression conditionExpression, AliasCollector collector) {
+    return ImmutableConditionExpression.builder()
+      .functions(conditionExpression.getFunctions().stream()
+        .map(f -> f.acceptExpressionFunction(this, collector)).collect(ImmutableList.toImmutableList()))
+      .build();
+  }
+
+  /**
+   * The callback method in the visitor design pattern.
+   * @param expressionFunction The object to be aliased.
+   * @param arguments constituent arguments of the object to be aliased.
+   * @param name the name of the object to be aliased.
+   * @param collector the class that does the aliasing.
+   * @return the aliased ExpressionFunction.
+   */
+  @Override
+  public ExpressionFunction visit(ExpressionFunction expressionFunction, List<Value> arguments,
```

Comment:
Can we update ExpressionFunction so we can get the other required fields via getters?

Reply:
Yes, as per your previous comments, this is my next task

Reply:
Now uses exposed getters directly.